### PR TITLE
Add support for interpreting nested and internal / external types

### PR DESCRIPTION
**Pull request by blacktooth**
This issue adds supports for parsing the nested references in the Proto schema correctly. This bug was raised as part of [issue](https://github.com/Apicurio/apicurio-registry/issues/1671). 

**Background**

When a nested reference is present in the schema, it is not interpreted correctly,

For example,
```
package foo.bar;

message A {
}

message  B {
    A a = 1;
}
```

When a `FileDescriptorProto` is generated for above schema, 

**Current behavior**

```
name: "fileName.proto"
package: "foo.bar"
message_type {
  name: "A"
  field {
  }
message_type {
  name: "B"
  field {
    name: "A"
    number: 1
    label: LABEL_OPTIONAL
    type: TYPE_MESSAGE
  }
}
```

**Expected behavior**

```
name: "fileName.proto"
package: "foo.bar"
message_type {
  name: "A"
  field {
  }
message_type {
  name: "B"
  field {
    name: "A"
    number: 1
    label: LABEL_OPTIONAL
    type: TYPE_MESSAGE
    type_name: ".foo.bar.A"

  }
}
```
In order to resolve the types of nested / internal types, square wire schema loader needs to be used instead of Proto parser [Reference](https://github.com/square/wire/issues/2024). 

**Summary of PR**

1. Create a `SchemaLoader` that can parse and link the Proto files.
2. Migrate `FileDescriptorUtils` to use `SchemaLoader` instead of just using `ProtoParser`.

**TODOs**

1. There is a [bug](https://github.com/square/wire/issues/2042) with square/wire library that prevents parsing schemas with no packages. Fix it here once it's fixed in the library.
2. The code here is lengthy, separate it into different modules. I didn't take it up in this PR to keep the diff easier to read.

**Testing**

1. Added unit tests that cover different scenarios.
2. Tests compare the generated `FileDescriptorProtos` against the one generated by Protobuf library.

### CODE REVIEW

**Comment by carlesarnal** [comment]
@blacktooth there are test failures, could you please take a look? Thanks!

**Comment by blacktooth** [comment]
Added missing dependencies for integration tests.

**Comment by blacktooth** [comment]
I ran into issues building this and running integration tests on an Amazon Linux ec2 instance. What is the recommended infrastructure to build and run the tests?

File: utils/protobuf-schema-utilities/src/test/proto/TestOrderingSyntax2References.proto [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package io.apicurio.registry.utils.protobuf.schema.syntax2.references;
```
Comment (hhkkxxx133) [inline utils/protobuf-schema-utilities/src/test/proto/TestOrderingSyntax2References.proto:1]: Can you also add an example proto for syntax 3 reference?

Comment (blacktooth) [reply utils/protobuf-schema-utilities/src/test/proto/TestOrderingSyntax2References.proto:1]: There is a reference case in `proto3`

Comment (hhkkxxx133) [reply utils/protobuf-schema-utilities/src/test/proto/TestOrderingSyntax2References.proto:1]: The test proto seems much simpler than this. Does internal/external also work for proto3?

Comment (blacktooth) [reply utils/protobuf-schema-utilities/src/test/proto/TestOrderingSyntax2References.proto:1]: Added detailed test cases now. Thanks

**Comment by carlesarnal** [comment]
> I ran into issues building this and running integration tests on an Amazon Linux ec2 instance. What is the recommended infrastructure to build and run the tests?

I don't think we have a recommended infrastructure to build the integration tests. That said, we're using ubuntu-18.04 in CI in case that helps.

**Comment by carlesarnal** [comment]
@blacktooth is this ready to merge?

**Comment by blacktooth** [comment]
Yes, this is ready to merge. Thanks!

**Review by carlesarnal — APPROVED** [review]
LGTM

**Comment by blacktooth** [comment]
Thanks a lot!

**Pull request merged**